main/asset_loaders/heatshrink_helper.c

```diff
@@ -81,3 +87,198 @@ uint8_t* readHeatshrinkFile(const char* fname, uint32_t* outsize, bool readToSpi
     // Return the decompressed bytes
     return decompressedBuf;
 }
+
+uint8_t* readHeatshrinkNvs(const char* namespace, const char* key, uint32_t* outsize, bool spiRam)
+{
+    // Read WSG from NVS
+    size_t sz;
+
+    // Get full size
+    readNamespaceNvsBlob(namespace, key, NULL, &sz);
+
+    ESP_LOGD("Heatshrink", "Compressed size is %" PRIu64, (uint64_t)sz);
+
+    uint8_t* buf = (uint8_t*)heap_caps_malloc(sz, spiRam ? MALLOC_CAP_SPIRAM : 0);
+    if (!buf)
+    {
+        return NULL;
+    }
+
+    if (!readNamespaceNvsBlob(namespace, key, buf, &sz))
+    {
+        free(buf);
+        return NULL;
+    }
+
+    // Pick out the decompresed size and create a space for it
```

Comment:
You should deduplicate the decompression. It seems to be shared whether it's coming from SPIFFS or NVS

Reply:
> You should deduplicate the decompression. It seems to be shared whether it's coming from SPIFFS or NVS

Ah yeah, I think I meant to do that at some point...